wgpu-hal/src/dx12/mod.rs

```diff
@@ -953,6 +953,7 @@ impl crate::DynPipelineLayout for PipelineLayout {}
 pub struct ShaderModule {
     naga: crate::NagaShader,
     raw_name: Option<ffi::CString>,
+    runtime_checks: bool,
```

Comment:
suggestion: It'd be good to document or rename so that it's clear that this means that runtime checks were requested for compilation. I don't think it's clear, just from the naming, but I understand now that I've been reading how it's used.

Reply:
The other backends also have this field, I'm not sure what it should be renamed to, do you have a suggestion?

Reply:
Maybe `inject_runtime_checks_for_indexing`?

Reply:
Oh, if this is already in other backends, then it's fine to be consistent as a first iteration. Might be interesting to clarify in follow-up work, though!

It's interesting that you reword the field as an imperative phrase. I thought it was indicative, i.e., "this was compiled with runtime checks". Maybe my understanding is wrong?

Reply:
The checks will be injected later when we call naga's backend. The module here is the IR generated by the WGSL frontend.